src/settings/App.tsx

```diff
@@ -47,13 +50,50 @@ export class App extends Component<{}, AppState> implements OC.Plugin<OC.Search.
 		sortOrder: 1
 	};
 
+	constructor(props) {
+		super(props);
+		// Enable the app if the user is member of a group with delegated admin rights
+		this.api.isGroupFoldersAdmin(OC.getCurrentUser().uid).then((isAdmin) =>  {
+			this.state.enabled = isAdmin;
+		})
+	};
+
 	componentDidMount() {
 		this.api.listFolders().then((folders) => {
 			this.setState({folders});
 		});
 		this.api.listGroups().then((groups) => {
 			this.setState({groups});
 		});
+		// Setup the admin delegation input field
+		this.api.listDelegatedAdmins().then((delegatedAdminGroups) => {
+			var t = this
+			var delegatedAdminsField = $('#groupfolders-root #groupfolders-admins-list');
+			delegatedAdminsField.val(delegatedAdminGroups);
+			delegatedAdminsField.select2({
+				data: this.state.groups.map(g => {
+					return { id: g.id, text: g.displayname };
+				}),
+				initSelection: function(element, callback) {
+					const selectedGroups = (delegatedAdminsField.val() as string).split('|').map(g => {
+						return { id: g, text: g }
+					})
+					callback(selectedGroups);
+				},
+				placeholder: 'Groups',
+				allowClear: true,
+				multiple: true,
+			});
+			delegatedAdminsField.change(function (event) {
+				const groups = (event.target as HTMLInputElement).value;
+				const uniqGroups = groups.split(/[,|]/).filter((value, index, self) => {
+					return self.indexOf(value) === index;
+				})
+				console.log('uniqGroups', uniqGroups);
+				// Persist changes in db
+				t.api.updateDelegatedAdmins(uniqGroups.toString());
+			});
+		});
```

Comment:
This is where the select2 issue occurs

Reply:
Any specific reason you went for select2 here? Apparently we already use react-select in the react code part of the app https://github.com/nextcloud/groupfolders/blob/6a81d9fc03fbd032a29a74d229e7b77e4f9dcf82/src/settings/FolderGroups.tsx#L106 for the group selection so I'd prefer to just stick with that for now. The GroupSelect component linked is probably rather straight forward to reuse.

Reply:
ok